diff -urN -x '*.orig' '-x*.rej' pytorch-1.4.0.orig/third_party/sleef/src/libm/CMakeLists.txt pytorch-1.4.0/third_party/sleef/src/libm/CMakeLists.txt
--- pytorch-1.4.0.orig/third_party/sleef/src/libm/CMakeLists.txt	2020-02-04 22:12:46.108000000 +0800
+++ pytorch-1.4.0/third_party/sleef/src/libm/CMakeLists.txt	2020-02-05 11:46:13.732000000 +0800
@@ -23,8 +23,7 @@
 endforeach()
 
 if(MSVC OR MINGW AND WIN32)
-  string(REPLACE "/" "\\" sleef_footer_input_file "${SLEEF_ORG_FOOTER}")
-  list(APPEND SLEEF_HEADER_COMMANDS COMMAND type ${sleef_footer_input_file} >> ${SLEEF_INCLUDE_HEADER})
+  list(APPEND SLEEF_HEADER_COMMANDS COMMAND cat ${SLEEF_ORG_FOOTER} >> ${SLEEF_INCLUDE_HEADER})
 else()
   list(APPEND SLEEF_HEADER_COMMANDS COMMAND cat ${SLEEF_ORG_FOOTER} >> ${SLEEF_INCLUDE_HEADER})
 endif()
